interpreter/src/cursor.rs

```diff
@@ -0,0 +1,3003 @@
+// Copyright (C) 2019-2024 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    cmp::Ordering,
+    collections::HashMap,
+    fmt,
+    hash::{Hash, Hasher},
+    mem,
+    str::FromStr as _,
+};
+
+use indexmap::{IndexMap, IndexSet};
+
+use rand::Rng as _;
+
+use rand_chacha::{ChaCha20Rng, rand_core::SeedableRng};
+
+use snarkvm::prelude::{
+    Address as SvmAddressParam,
+    Boolean as SvmBooleanParam,
+    Cast,
+    CastLossy as _,
+    Double as _,
+    Field as SvmFieldParam,
+    FromBits as _,
+    Group as SvmGroupParam,
+    Identifier as SvmIdentifierParam,
+    Inverse as _,
+    Network as _,
+    Pow as _,
+    Scalar as SvmScalarParam,
+    // Signature as SvmSignatureParam,
+    SizeInBits,
+    Square as _,
+    SquareRoot as _,
+    TestnetV0,
+    ToBits,
+    integers::Integer as SvmIntegerParam,
+};
+
+type SvmAddress = SvmAddressParam<TestnetV0>;
```

Comment:
Do we want these types parameterized over the `Network`?

Reply:
We just rolled some updates where `TestnetV0` and `MainnetV0` can differ in behavior depending on the block height. I could see that as being a potential case, but not much else.